storage/boltdb/client.go

```diff
@@ -112,13 +115,34 @@ func (client *Client) view(fn func(*bolt.Bucket) error) error {
 	}))
 }
 
-// Put adds a value to the provided key in boltdb, returning an error on failure.
+// Put adds a key/value to BoltDB in a batch, where boltDB commits the batch to
+// to disk every 1000 operations or 10ms, whichever is first.
+// The MaxBatchDelay are using default settings and be changed if need be.
+// Ref: https://github.com/boltdb/bolt/blob/master/db.go#L160
+// Note: when using this method, make sure its being executed asynchronously since
+// it blocks for the duration db.MaxBatchDelay.
 func (client *Client) Put(key storage.Key, value storage.Value) error {
+	start := time.Now()
 	if key.IsZero() {
 		return storage.ErrEmptyKey.New("")
 	}
 
-	return client.update(func(bucket *bolt.Bucket) error {
+	err := client.db.Batch(func(tx *bolt.Tx) error {
+		bucket := tx.Bucket(client.Bucket)
+		return bucket.Put(key, value)
+	})
+	mon.IntVal("boltDB Batch time elapsed").Observe(int64(time.Since(start)))
+	return err
+}
+
+// PutAndCommit adds a key/value to BoltDB and writes it to disk.
+func (client *Client) PutAndCommit(key storage.Key, value storage.Value) error {
+	if key.IsZero() {
+		return storage.ErrEmptyKey.New("")
+	}
+
+	return client.db.Update(func(tx *bolt.Tx) error {
```

Comment:
Is there a particular reason not to stick with `client.update()` here, as the other Client methods do? If no, probably best to switch back to avoid confusion. If yes, this deserves an explanatory comment, and if the same reason applies to the other methods, maybe those should be changed as well.

Reply:
i thought the way it was written before was confusing.  For example:

```
// PutAndCommit adds a value to the provided key in boltdb, returning an error on failure.
func (client *Client) PutAndCommit(key storage.Key, value storage.Value) error {
	if key.IsZero() {
		return storage.ErrEmptyKey.New("")
	}

	return client.update(func(bucket *bolt.Bucket) error {
		return bucket.Put(key, value)
	})
}

func (client *Client) update(fn func(*bolt.Bucket) error) error {
	return Error.Wrap(client.db.Update(func(tx *bolt.Tx) error {
		return fn(tx.Bucket(client.Bucket))
	}))
}
```

I felt like we could remove the `update` method all together, but now that Im looking at it more, I guess its written like that to abstract away the boltDB method calls. I'm not sure. Anyhow, I can revert it to keep it consistent.

Reply:
Yeah, I'm not sure what the reasoning is behind the `update()` method. Probably to abstract those particular parts of Bolt world. But yeah, consistency helps readability a lot on its own. We should either use `update()` here or take it out in the other methods (and if you want to do that, a separate commit would probably be preferable).

Reply:
i think i will revert the changes in this PR, then submit a different PR to remove that layer of abstraction i.e `update`, etc methods